server/controllers/file.controller.js

```diff
@@ -7,68 +7,90 @@ import { deleteObjectsFromS3, getObjectKey } from './aws.controller';
 // Bug -> timestamps don't get created, but it seems like this will
 // be fixed in mongoose soon
 // https://github.com/Automattic/mongoose/issues/4049
-export function createFile(req, res) {
-  Project.findOneAndUpdate(
-    {
-      _id: req.params.project_id,
-      user: req.user._id
-    },
-    {
-      $push: {
-        files: req.body
+export async function createFile(req, res) {
+  try {
+    const updatedProject = await Project.findOneAndUpdate(
+      {
+        _id: req.params.project_id,
+        user: req.user._id
+      },
+      {
+        $push: {
+          files: req.body
+        }
+      },
+      {
+        new: true
       }
-    },
-    {
-      new: true
-    }, (err, updatedProject) => {
-      if (err || !updatedProject) {
-        console.log(err);
-        res.status(403).send({ success: false, message: 'Project does not exist, or user does not match owner.' });
+    );
+
+    if (!updatedProject) {
+      res.status(403).send({
+        success: false,
+        message: 'Project does not exist, or user does not match owner.'
+      });
+      return;
+    }
+    const newFile = updatedProject.files[updatedProject.files.length - 1];
+    updatedProject.files.id(req.body.parentId).children.push(newFile.id);
+    updatedProject.save(innerErr => {
+      if (innerErr) {
+        console.log(innerErr);
+        res.json({ success: false });
         return;
       }
-      const newFile = updatedProject.files[updatedProject.files.length - 1];
-      updatedProject.files.id(req.body.parentId).children.push(newFile.id);
-      updatedProject.save((innerErr) => {
-        if (innerErr) {
-          console.log(innerErr);
-          res.json({ success: false });
-          return;
-        }
-        res.json(updatedProject.files[updatedProject.files.length - 1]);
+      res.json(updatedProject.files[updatedProject.files.length - 1]);
+    });
+  } catch (err) {
+    if (err) {
+      res.status(403).send({
+        success: false,
+        message: 'Project does not exist, or user does not match owner.'
       });
+      return;
     }
-  );
+  }
 }
 
 function getAllDescendantIds(files, nodeId) {
   const parentFile = files.find(file => file.id === nodeId);
   if (!parentFile) return [];
-  return parentFile.children
-    .reduce((acc, childId) => (
-      [...acc, childId, ...getAllDescendantIds(files, childId)]
-    ), []);
+  return parentFile.children.reduce(
```

Comment:
Just as a heads up, these stylistic changes should really not be included, it breaks the git blame for this line and is generally unhelpful when looking at diffs.

That said, making an additional commit to revert them would be worse in general, so only fix that if you're going to be doing it by altering commit history. I just thought I'd point out that it is something to look out for when you're creating a PR: If you have diffs on lines irrelevant to what you're doing, something usually has gone wrong :)

Reply:
My global prettier generally makes these changes , since the project does not has any local .prettierrc file , my global prettier kicks in and make the changes

Reply:
for this project, i squash commits for a PR, so i'm okay with you making commits to revert stuff. i also agree, these stylistic changes should not be included in this PR.